parser/prism/Translator.cc

```diff
@@ -196,7 +196,12 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             auto backReferenceReadNode = down_cast<pm_back_reference_read_node>(node);
             auto name = translateConstantName(backReferenceReadNode->name);
 
-            return make_unique<parser::Backref>(location, name);
+            auto locZeroLen = location.copyWithZeroLength();
+            auto recv = MK::Magic(location);
+            auto arg = MK::Symbol(location, name);
+            auto expr = MK::Send1(location, move(recv), core::Names::regexBackref(), locZeroLen, move(arg));
```

Comment:
For cases like this, where we're synthesizing fake method calls, let's add a Ruby-like comment to helper readers understand what we're desugaring to:


```suggestion
            // Desugar `$foo` to `<Magic>.<regex-backref>(:foo)`
            auto recv = MK::Magic(location);
            auto arg = MK::Symbol(location, name);
            auto expr = MK::Send1(location, move(recv), core::Names::regexBackref(), locZeroLen, move(arg));
```